src/api/app/helpers/webui/request_helper.rb

```diff
@@ -139,4 +142,47 @@ def diff_data(action_type, sourcediff)
   def diff_label(diff)
     "#{diff['project']} / #{diff['package']} (rev #{diff['rev']})"
   end
+
+  # rubocop:disable Style/FormatStringToken, Style/FormatString
+  def request_action_header(action, creator)
+    source_project_hash = { project: action[:sprj], package: action[:spkg] }
+
+    case action[:type].to_sym
```

Comment:
I like this one - I split this into partials per action type, but this makes changes quite hard

Reply:
tbh, I am still not 100% happy with it. But I guess it's good enough for now.